December 2019 update

ngambino0192 opened the pull request:
Initial integration checkpoint. Flagging this branch as `Ready for Review`

Lots of changed files, so I added three comments to files that I think need particular attention. @lukebaker if you sign off on this review, I can also update the `README.md` as there is a slightly different approach to some of the frontmatter needed for rendering out new locations (this is where most of the changed files are).

Also to note, I am waiting on new images for the detail pages and location pages. As this is a minor addition, I decided to go ahead and move forward with the PR.

ngambino0192 commented on _includes/sidebar.html:
```diff
@@ -0,0 +1,132 @@
+    </ul>
+    <ul class="nav flex-column">
+      <a class="nav-link collapsed" onmouseover=""" style="cursor: pointer;" data-toggle="collapse" data-target="">Other Events:</a>
+      {% assign check2018 = site.data.2018[location].hero %}
```
Not the most elegant solution to the "See events from other years" functionality. Any suggestions?

lukebaker replied:
Yes. I did something similar here:

https://github.com/compsocialscience/summer-institute/blob/4438440d548c65e2af6852d113d7c94b6f234112/_layouts/all_time_participants.html#L13-L14

You can use that to get a list of all possible years and then check to see if that location existed for each year.

ngambino0192 replied:
thanks! addressed in: https://github.com/compsocialscience/summer-institute/pull/1240/commits/cb3ce84ada63ee46111643ca58e93e6580da1e51

ngambino0192 commented on _includes/sidebar.html:
```diff
@@ -0,0 +1,132 @@
+      <li class="nav-item">
+        <a class="nav-link" onmouseover="" style="cursor: pointer;" data-toggle="collapse" data-target="#collapse{{forloop.index}}">{{year}}</a>
+      </li>
+      <ul class="nav flex-column collapse" id="collapse{{forloop.index}}" aria-labelledby="{{forloop.index}}" data-parent="#accordionMenu">
```
this seems to mess up the styles when the accordian menu is transitioning. I think this might be a bootstrap issue, something to do with the `.collapsing` style. https://getbootstrap.com/docs/4.0/components/collapse/#example

ngambino0192 replied:
addressed in: https://github.com/compsocialscience/summer-institute/pull/1240/commits/b997acb2e3f32fb2a09b23af0aa086c207cb6541

ngambino0192 commented on _includes/curriculum_listing.html:
```diff
@@ -0,0 +1,52 @@
+    {% if curriculum.events %}
+      {% for event in curriculum.events %}
+        {% if event %} 
+        <div class="col-lg-12">
```
card borders aren't closing on the right and bottom borders. another bootstrap issue that I believe may be related to the v4 version we are using?

ngambino0192 replied:
addressed in: https://github.com/compsocialscience/summer-institute/pull/1240/commits/6bac5eed94e153bdb5256d50da58c8721eeaddd4

lukebaker commented on 2019/bamberg/schedule.md:
```diff
@@ -0,0 +1,7 @@
+---
+layout: location_detail
+partner_site: capetown
```
This doesn't match the directory it is in. Is this incorrect?

ngambino0192 replied:
addressed in: https://github.com/compsocialscience/summer-institute/pull/1240/commits/ea27995cfeab1f83e8100a17a482be32ba26a1e2

lukebaker commented on 2019/capetown/schedule.md:
```diff
@@ -0,0 +1,7 @@
+---
+layout: location_detail
+partner_site: bramberg
```
This doesn't match the directory it is in. Is this incorrect?

ngambino0192 replied:
addressed in: https://github.com/compsocialscience/summer-institute/pull/1240/commits/45c69738e5def1947158684399c06d8a7735dc0d

lukebaker commented on _data/2020/locations.yml:
```diff
@@ -0,0 +1,168 @@
+- year: 2020
```
Why does this live in the `2020` directory but include all year locations? Here are three other options that I think are better (with their own trade-offs):
1. One global `locations.yml` file in `_data/`.
2. A separate `locations.yml` file for each year that only contains location data for that year. (Less likely that folks will mess up prior years).
3. Each location has their own `location.yml` with this data. This is slightly harder to iterate through (you could do it with [this technique](https://github.com/compsocialscience/summer-institute/blob/4438440d548c65e2af6852d113d7c94b6f234112/_layouts/all_time_participants.html#L13-L22), but makes it more likely that any changes will only mess up one location.

ngambino0192 replied:
Went with recommendation `#3`. Was a bit trickier to implement than I initially thought, but the new locations page and it's corresponding sidebar has been implemented in: https://github.com/compsocialscience/summer-institute/pull/1240/commits/2b0e05f08bb62377a87b86e78e6fad449f4ca8e8 and https://github.com/compsocialscience/summer-institute/pull/1240/commits/a461342bc65c191e50644eafa1d391c427c52568

ngambino0192 commented on 2019/bamberg/apply.md:
```diff
@@ -1,9 +1,13 @@
+partner_site: bramberg
+layout: location_detail
+title: Bramberg
+location: University of Bramberg
```
addressed in: https://github.com/compsocialscience/summer-institute/pull/1240/commits/d14a5b2cbf93d48f2017b77bbc07352588b90cf7

ngambino0192 commented on _data/2018/capetown/navigation.yml:
```diff
@@ -1,10 +1,8 @@
+- name: Locations
+  url: "/2020/locations"
+- name: Curriculum
+  url: "/2020/curriculum"
```
I agree. Addressed in: https://github.com/compsocialscience/summer-institute/pull/1240/commits/c67045e611526f25fd833b00265bc16377b315a2

ngambino0192 commented:
> @ngambino0192 Thanks for jumping on this one. I accidentally submitted a bunch of comments before this one. There are a few items that need changing. Feel free to ask me any questions.
> 
> * 2020 Northwestern has lots of references to "Bramberg"
> * [Links on the dark sections are unreadable.](https://user-images.githubusercontent.com/84748/70457843-e0ec3780-1a7e-11ea-9fdf-4654a37c8a89.png)
> * /summer-institute/2018/boulder/ looks a bit funny at the bottom with the [white stripe](https://user-images.githubusercontent.com/84748/70458038-46402880-1a7f-11ea-8e6e-1638ce9dd284.png)
> * It seems like the "primary location" for each year isn't showing up. For example, where can I find the content that currently exists on this URL: https://compsocialscience.github.io/summer-institute/2018/. We might need to convert those to be a proper location on their just like the rest of the locations rather than at the level that they currently exist at.

Putting this PR back into a second round of review. I tried to address each of your comments as well as I could, but let me know if/where this can be improved.

ngambino0192 commented:
> @ngambino0192 I added several commits to change / simplify things. It would be good to review those. The biggest change is moving the primary (the things linked in the top nav) up a level so that they're no longer in the 2020 directory. Those pages are year agnostic and should live outside of the year structure.
> 
> If you could take care of these additional changes, that'd be great:
> 
> 1. The "People" page from the top navbar should show all people ever. There's an [existing page on the site](https://compsocialscience.github.io/summer-institute/participants) that shows all the people. You can use that [as a guide](https://github.com/compsocialscience/summer-institute/blob/4438440d548c65e2af6852d113d7c94b6f234112/_layouts/all_time_participants.html#L13-L22) for how to fetch all people ever.
> 2. Can you review the primary location data. I noticed that `2018/duke/index.md` seems to have the same content as `2020/duke/index.md`.
> 3. Now that we no longer have a "year home" (e.g., /2020/) perhaps those URLs should redirect to the locations page for that year: `/locations#2020`.
> 4. Update documentation for creating a new partner location. Removing anything that doesn't apply anymore.
> 5. Remove "Lorem ipsum" text. Empty is probably better in most cases.

These issues have been addressed. I took a slightly different approach to linking the `/locations#2020` we spoke about briefly in slack. I'm creating a new `index.md` file that defines the redirection and a `redirect.html` layout that actually handles the redirect. 

https://stackoverflow.com/questions/39657226/jekyll-redirect-link

Also updated the README. I tried to keep it to a similar scope as what was there before.

ngambino0192 commented:
Thinking that this is nearly there. Can you have a look at the README and let me know what you think?
